### DISCUSSION TITLE
Make color picker default colors more "internal / theme agnostic"

### DISCUSSION THREAD

**Discussion by LasseRosenow** [directus/directus#12969] [Feature Requests]
So right now the color picker, besides being able to insert what ever hex code you'd like allows you to insert some default color values as well:

![image](https://user-images.githubusercontent.com/10547444/164837819-925f0545-7501-4aba-b2b7-acad06ad659e.png)

So the rough idea is that those colors should no longer just insert hex values, but css variables or something like that, maybe even something in the likes of how it is handled with the `translation strings` right now.

The benefit would be that those default colors could be made adaptive to the system theme. They could be made more bright in dark mode for example to have a better contrast. Also they could be `themed`.

Especially `white` and `black` are a problem if you use light or dark mode right now.

I think the new selection could be something like this: (note the black white thing)

![image](https://user-images.githubusercontent.com/10547444/164838153-f0613d10-cb56-4144-bf29-6892d5ff4cc9.png)

(Of course we could also show each color the same way, but I thought it might only be necessary for black and white.)

And it could then be visualised somewhat like this:

![image](https://user-images.githubusercontent.com/10547444/164838389-42e6702f-c349-4f32-a80f-f194145aa2c8.png)

**Comment by benhaynes** [comment]
Absolutely... this is something that we're working on now with @AustinPhillipTaylor. Should be part of our broader Theme System sprint. 👍

**Comment by schroeder-g** [comment]
Has this feature been implemented? I'm looking for a way to set a project default for the color-picker

**Comment by schroeder-g** [comment]
It looks like preset colors are not being injected into the color wheel correctly; 

Right now I have a `Brand Colors` model with a single `hex` field. I have preset colors like so:
![image](https://user-images.githubusercontent.com/66321261/214387762-0469465d-6fb5-46ea-8efc-39bc9fa498e6.png)

However, when I try to create a new instance of the Brand Colors with one of those presets, none appear below the color picker:
![image](https://user-images.githubusercontent.com/66321261/214387942-0ad87d28-e433-4ff8-a7f1-71de2bfbc5ea.png)

How can I leverage the color presets? Any input is appreciated 🙏

**Reply by LasseRosenow** [reply]
What you see there is the system color picker. You need to click inside the field not on the color in the field, to open the directus color picker

**Reply by schroeder-g** [reply]
Thanks you Lasse!

**Reply by schroeder-g** [reply]
I suppose the follow up question would be: how do I make my color presets available on all color picker fields?

**Reply by LasseRosenow** [reply]
That I don't know. I fear it is not possible 😭

**Reply by LasseRosenow** [reply]
But I would recommend you to open a new discussion for this question, since it is kinda out of scope regarding what this discussion is about :)

**Reply by rijkvanzanten** [reply]
There's currently no way to set global interface defaults, but you can copy/paste the configuration between fields, or duplicate the field to other collections to speed up the config process a bit 🙂

**Reply by benhaynes** [reply]
This warrants a new/dedicated Discussion for a feature request, but this conversation just gave me the idea of open-ended "Global Variables" for the system. Much like our "Global Translation Variables" which can be used anywhere in the system, it'd be pretty powerful to allow users to define a global var (eg: `primary_color` = `#6644FF`, or `category_3` = `Finance`, etc) and then they can be used in the interface options or elsewhere. Maybe that gets out of hand, but interesting idea.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
